src/Files.App/Helpers/AppLifecycle.cs

```diff
@@ -0,0 +1,360 @@
+						SharedMemoryName = SharedMemoryNameHeader + defaultBufferSize.ToString();
+					}
+				}
+			}
+			catch (FileNotFoundException)
+			{
+				SharedMemoryName = SharedMemoryNameHeader + defaultBufferSize.ToString();
+				SharedMemoryNameHeaderMemory = MemoryMappedFile.CreateOrOpen(SharedMemoryNameHeader, 1024);
+				using (MemoryMappedViewAccessor accessor = SharedMemoryNameHeaderMemory.CreateViewAccessor())
+				{
+					byte[] buffer = Encoding.UTF8.GetBytes(SharedMemoryName);
+					accessor.WriteArray(0, buffer, 0, buffer.Length);
+				}
+			}
+			return SharedMemoryNameHeaderMemory;
+		}
+
+
+
+		//	Check if the SharedMemory exists, if not create it
+		protected static MemoryMappedFile CheckSharedMemory()
+		{
+			GetSharedMemoryNameHeader();
+			try
+			{
+				SharedMemory = MemoryMappedFile.OpenExisting(SharedMemoryName);
+			}
+			catch (FileNotFoundException)
+			{
+				SharedMemory = MemoryMappedFile.CreateOrOpen(SharedMemoryName, defaultBufferSize);
+			}
+			return SharedMemory;
+		}
+
+		//	Check if the SharedMemory exists and if the BufferSize is enough, if not create a new one
+		protected static MemoryMappedFile CheckSharedMemory(int BufferSize)
+		{
+			SharedMemory = CheckSharedMemory();
+			long BufferSizeIn = BufferSize;
+			using (MemoryMappedViewAccessor accessor0 = SharedMemory.CreateViewAccessor())
+			{
+				long length = accessor0.Capacity;
+				if (length > BufferSizeIn)
+				{
+				}
+				else
+				{
+					SharedMemory.Dispose();
+					long NewBufferSize = ((BufferSizeIn / defaultBufferSize) + 1) * defaultBufferSize;
+					SharedMemoryName = SharedMemoryNameHeader + NewBufferSize.ToString();
+					SharedMemory = MemoryMappedFile.CreateOrOpen(SharedMemoryName, NewBufferSize);
+					SharedMemoryNameHeaderMemory = MemoryMappedFile.CreateOrOpen(SharedMemoryNameHeader, 1024);
+					using (MemoryMappedViewAccessor accessor1 = SharedMemoryNameHeaderMemory.CreateViewAccessor())
+					{
+						byte[] buffer = Encoding.UTF8.GetBytes(SharedMemoryName);
+						accessor1.WriteArray(0, buffer, 0, buffer.Length);
+					}
+				}
+			}
+			return SharedMemory;
+		}
+
+		//	Read TabsWithIDArgList from SharedMemory
+		protected static async Task ReadSharedMemory()
+		{
+			try
+			{
+				SharedMemory = CheckSharedMemory();
+				using (MemoryMappedViewAccessor accessor = SharedMemory.CreateViewAccessor())
+				{
+					long length = accessor.Capacity;
+					byte[] buffer = new byte[length];
+					accessor.ReadArray(0, buffer, 0, buffer.Length);
+					int nullIndex = Array.IndexOf(buffer, (byte)'\0');
+					if (nullIndex > 0)
+					{
+						int Index = nullIndex;
+						byte[] truncatedBuffer = new byte[Index];
+						Array.Copy(buffer, 0, truncatedBuffer, 0, Index);
+						string bufferStr = Encoding.UTF8.GetString(truncatedBuffer);
+						List<string> TabsWithIDArgStrList = JsonSerializer.Deserialize<List<string>>(bufferStr);
+						TabsWithIDArgList = TabsWithIDArgStrList.Select(x => TabItemWithIDArguments.Deserialize(x)).ToList();
```

Comment:
```suggestion
						List<string> TabsWithIDArgStrList = JsonSerializer.Deserialize<List<string>>(bufferStr)
																										.Select(x => TabItemWithIDArguments.Deserialize(x))
																										.ToList();
```

Reply:
We can eliminate the need for `TabsWithIDArgStrList` since it's only used directly under.